File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height) - 1
        max_area = 0

        while i < j:
            width = j - i
            min_height = min(height[i], height[j])
            current_area = width * min_height
            max_area = max(current_area, max_area)

            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
            
        return max_area

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0, j = height.length - 1;
        int max_area = 0;

        while (i < j){
            int width = j - i;
            int min_height = Math.min(height[i], height[j]);
            int current_area = width * min_height;
            max_area = Math.max(max_area, current_area);

            if (height[i] < height[j])
                i++;
            else
                j--;
        }

        return max_area;

    }
}